java-bootcamp-2018-tdd/src/main/java/com/globant/dbConnection/SQLConnection.java

```diff
@@ -0,0 +1,387 @@
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+	}
+
+	/**
+	 * statement and connection
+	 */
+
+	public void close() {
+		try {
+			if (statement != null) {
+				statement.close();
+			}
+			if (connect != null) {
+				connect.close();
+			}
+		} catch (Exception e) {
+		}
+	}
+
+	/**
+	 * search a user by nickname in DB
+	 * 
+	 * @param nickName
+	 * @return a user
+	 * @throws SQLException
+	 */
+	public UserModel findUser(String nickName) throws SQLException {
+		ResultSet resultSet = statement.executeQuery("SELECT U.nickName, U.firstName, U.lastName, U.password"
+				+ " FROM Users U" + " WHERE U.nickName='" + nickName + "'");
+		UserModel user = new UserModel();
+		if (resultSet.next()) {
+			user.setNickName(resultSet.getString(NICKNAME));
+			user.setFirstName(resultSet.getString(FIRSTNAME));
+			user.setLastName(resultSet.getString(LASTNAME));
+			user.setPassword(resultSet.getString(PASSWORD));
+		}
+		resultSet.close();
+		return user;
+	}
+
+	/**
+	 * find a saved shopping cart of a user in DB
+	 * 
+	 * @param nickName
+	 * @return a shopping cart
+	 * @throws SQLException
+	 */
+	public IShoppingCart findCart(String nickName) throws SQLException {
+		ResultSet resultSet = statement.executeQuery("SELECT R.name, R.quantity"
+				+ " FROM Carts C INNER JOIN CartRows R ON C.id=R.id" + " WHERE C.nickName='" + nickName + "'");
+		IShoppingCart shoppingCart = new ShoppingCart();
+		List<ProductModel> products = new ArrayList<ProductModel>();
+		while (resultSet.next()) {
+			ProductModel prod = new ProductModel();
+			prod.setName(resultSet.getString(NAME));
+			prod.setQuantity(resultSet.getInt(QUANTITY));
+			products.add(prod);
+		}
+		resultSet.close();
+		for (ProductModel product : products) {
+			ProductModel toAdd = findProduct(product.getName());
+			shoppingCart.addProduct(product.getName(), toAdd.getCategory(), toAdd.getPrice(), product.getQuantity());
+		}
+		return shoppingCart;
+	}
+
+	/**
+	 * search a product for name in the DB
+	 * 
+	 * @param name
+	 * @return a product
+	 * @throws SQLException
+	 */
+	public ProductModel findProduct(String name) throws SQLException {
+		ResultSet resultSet = statement.executeQuery(
+				"SELECT P.name, P.category, P.price" + " FROM Products P" + " WHERE P.name='" + name + "'");
+		ProductModel product = new ProductModel();
+		if (resultSet.next()) {
+			product.setName(name);
+			product.setPrice(resultSet.getDouble(PRICE));
+			product.setCategory(resultSet.getString(CATEGORY));
+		}
+		resultSet.close();
+		return product;
+	}
+
+	/**
+	 * search a cart Id of a user
+	 * 
+	 * @param nickName
+	 * @return a cartId
+	 * @throws SQLException
+	 */
+	public int findCartId(String nickName) throws SQLException {
+		ResultSet resultSet = statement
+				.executeQuery("SELECT C.id" + " FROM Carts C" + " WHERE C.nickName='" + nickName + "'");
+		if (resultSet.next()) {
+			return resultSet.getInt(ID);
+		}
+		resultSet.close();
+		return -1;

```

Comment:
When you implement a custom error you should document it on the javadoc.
You can also return Integers so you can return a null, or maybe implement a custom exception